### ISSUE TITLE
[BUG] gridplot reset button does not reset all figures 

### ISSUE THREAD

**Issue by Hoxbro** [bokeh/bokeh#12578]
### Software versions

Python version      :  3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:58:50) 
IPython version     :  8.6.0
Tornado version     :  6.2
Bokeh version       :  3.0.1
BokehJS static path :  /home/shh/miniconda3/envs/bokeh_clean/lib/python3.9/site-packages/bokeh/server/static
node.js version     :  (not installed)
npm version         :  (not installed)
Operating system    :  Linux-6.0.2-76060002-generic-x86_64-with-glibc2.35

### Jupyter notebook / Jupyter Lab version

Jupyter lab 3.5.0

### Expected behavior

When using `gridplot`, the reset button should reset all the figures, not only one.  

### Observed behavior

It only resets one of the figures. This used to work as expected in Bokeh 2.4.3.

### Example code

```Python
from bokeh.io import output_notebook, show
from bokeh.layouts import gridplot
from bokeh.plotting import figure

output_notebook()


def fig():
    fig = figure(height=200, width=200)
    a = fig.line(x=list(range(10)), y=list(range(10)))
    return fig


show(gridplot([[fig(), fig()]]))
```

### Screenshots

Bokeh 2.4.3:

https://user-images.githubusercontent.com/19758978/200530994-2fb52e52-58c0-4b7e-91b5-303b14a36769.mp4

Bokeh 3.0.1:

https://user-images.githubusercontent.com/19758978/200531014-6c527676-0638-48b8-b8d9-35d96557561f.mp4

**Comment by mattpap** [comment]
Applies to all action tools that act on individual plots.

**Comment by jgborch** [comment]
Hello, I'm a computer science student at the University of Michigan, and I was looking to contribute for a course project. This would be my first contribution to an open-source project, so I'm wondering if this would be a good bug-fix to take on?

**Comment by bryevdv** [comment]
Hi @jgborch there has already been a PR submitted with a fix by @mattpap (#12580) Can you open a [GitHub discussion](https://github.com/bokeh/bokeh/discussions) and introduce yourself / describe your background and interests a little? Then we can look for an issue that could be a good fit.
